src/authservice/chart/templates/authn.yaml

```diff
@@ -0,0 +1,20 @@
+# Authservice is non-functional without Istio/RequestAuthentication but we wrap this in a conditional to handle standalone testing
+{{- if .Capabilities.APIVersions.Has "security.istio.io/v1beta1" }}
+apiVersion: security.istio.io/v1beta1
+kind: RequestAuthentication
+metadata:
+  name: jwt-authn
+  namespace: istio-system
+spec:
+  selector:
+    matchLabels:
+      {{ .Values.selector.key }}: {{ .Values.selector.value | quote }}
+  jwtRules:
```

Comment:
I always wanted to include `audiences:` in this, but it would need to be done per namespace and not a global istio-system policy. This Helm chart was originally written to kind of do everything and made some compromises.

Currently this RequestAuthentication  policy allows any JWT from the correct issuer and does not check the audience `aud` claim (the Keycloak client ID). In theory with something like an XSS attack, you could steal one app's JWT and use it to authenticate do a different app.

In practice, many apps don't send the JWT down to the client... but some do. Also Authservice now strips the client-provided `Authentication:` header... but there have been ways to bypass Authservice entirely in the past. I don't consider these perfect mitigations and would prefer per client per namespace enforcement done in Istio.

Have we considered something like requesting Authservice protection in the UDS CR? Then Pepr could generate the RequestAuthentication per-namespace with a specific Client ID.

Reply:
Definitely would trust your judgement and knowledge here way more than mine. Assuming a follow on PR handles client creation + wiring to authservice + creates a per client `RequestAuthentication`...seem wise to remove this global one from the chart now?

Reply:
Whichever, remove now, leave it and opt out, or remove later. We can leave this if you want. Apps can opt out of the global policy by not using the `protect: keycloak` label or whatever it's configured to. If they don't match they label they don't get the AuthorizationPolicies either, but I think if you're doing per-namespace RequestAuthentication, then per-namespace AuthorizationPolicies may make sense too anyway?

Reply:
Leaning towards leaving until we know better what the follow on / replacement looks like. Not sure it makes a big difference either way since keycloak isn't here yet but I think this is a good callout for the next PR for sure. Going to resolve this thread and keep it in mind for IDAM PR to follow :+1: